github_slack_bot/utils/slack.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import re
+import time
+from slackclient import SlackClient
+from slackclient.client import SlackNotConnected
+from slackclient.server import SlackLoginError
+
+logger = logging.getLogger()
+
+RTM_READ_INTERVAL = 1  # Number of seconds to wait between each rtm_read() call
+EXCEPTION_DEFAULT_MSG = "Something unexpected occurred, see ERROR log and Traceback above"
+
+
+class SlackBot:
+    """The SlackBot listens to events from the Slack RTM API and creates GitHub issues
+    from messages that mention the bot's username.
+    """
+
+    # TODO: extend is_bot_message() to allow bot mentions anywhere in the message
+    @staticmethod
+    def is_bot_mention(bot_user_id, message):
+        """Check if the message mentions the bot_user_id at
+        the start of the message and includes a body
+        :param bot_user_id: bot's user ID to search for
+        :param message: message sent by user
+        :return: True if bot is mentioned and message contains body, False otherwise
+        """
+        message_stripped = message.strip()
+        logger.debug("Checking if bot '{}' is mentioned in message: '{}'"
+                     .format(bot_user_id, message_stripped))
+        return message_stripped.startswith("<@{}> ".format(bot_user_id))
```

Comment:
So in order for the bot to detect the message, it looks like you will have to mention it using its "slack id"? I.e. if I join it to a channel with name, say, "ticketbot" - will I be able to say `@ticketbot create-ticket blah` or do I need to find out its id and use that?

Reply:
To message the bot you just tag it normally like "@ticketbot", but under the hood Slack delivers the message through the API with that tagged user's ID (e.g. <@U3IE83803>).